packages/prop-controllers/src/width/width.test.ts

```diff
@@ -0,0 +1,134 @@
+import { ControlDataTypeKey, Types } from '../prop-controllers'
+import {
+  WidthDescriptor,
+  WidthPropControllerDataV0,
+  WidthPropControllerDataV1,
+  WidthPropControllerDataV1Type,
+  ResponsiveWidthData,
+  createWidthPropControllerDataFromResponsiveWidthData,
+  getWidthPropControllerDataResponsiveWidthData,
+} from './width'
+
+describe('WidthPropController', () => {
+  describe('getWidthPropControllerDataResponsiveWidthData', () => {
+    test('returns value for WidthPropControllerDataV1Type', () => {
+      // Arrange
+      const data: WidthPropControllerDataV1 = {
+        [ControlDataTypeKey]: WidthPropControllerDataV1Type,
+        value: [
+          {
+            deviceId: 'desktop',
+            value: { value: 17, unit: 'px' },
+          },
+        ],
+      }
+
+      // Act
+      const result = getWidthPropControllerDataResponsiveWidthData(data)
+
+      // Assert
+      expect(result).toEqual(data.value)
+    })
+
+    test('returns value for WidthPropControllerDataV0 data', () => {
+      // Arrange
+      const data: WidthPropControllerDataV0 = [
+        {
+          deviceId: 'desktop',
+          value: { value: 17, unit: 'px' },
+        },
+      ]
+
+      // Act
+      const result = getWidthPropControllerDataResponsiveWidthData(data)
+
+      // Assert
+      expect(result).toEqual(data)
+    })
+
+    test('returns value for unknown data', () => {
+      // Arrange
+      const data = {
+        test: 'unknown',
+      }
+
+      // Act
+      // @ts-expect-error: invalid data
+      const result = getWidthPropControllerDataResponsiveWidthData(data)
+
+      // Assert
+      expect(result).toEqual(data)
+    })
```

Comment:
See my other comment on weird test cases like this.

Reply:
Removed